folly/container/test/F14MapTest.cpp

```diff
@@ -1365,14 +1365,20 @@ TEST(F14ValueMap, maxSize) {
   F14ValueMap<int, int> m;
   EXPECT_EQ(
       m.max_size(),
-      std::numeric_limits<std::size_t>::max() / sizeof(std::pair<int, int>));
+      std::min(
```

Comment:
From @nbronson,

> We don't need `numeric_limits` here at all.

(Or the `min`.)

Reply:
Oh, I misunderstood a few things in `F14Table` and `F14BasicMap`.

If I understand now correctly, the `max_size()` of `F14ValueMap` is the min of the following 2 components:
- the maximum value that the type `F14Table::InternalSizeType` can represent
- the `max_size()` of `F14Table`'s allocator.

Since the allocator for a simple `F14ValueMap` is `DefaultAlloc`, which is `std::allocator<T>`,
and because `BasePolicy::InternalSizeType` is `size_t`, the correct assertion is:

```C++
EXPECT_EQ(
      m.max_size(),
      std::allocator_traits<decltype(m)::allocator_type>::max_size(m.get_allocator())
)
```

Reply:
Should I be more explicit? For example:

```C++
  F14ValueMap<int, int> m;

  std::allocator<std::pair<int, int>> alloc;
  EXPECT_EQ(
      m.max_size(),
      std::allocator_traits<decltype(alloc)>::max_size(alloc));
```

Reply:
The first form is fine; don't need to be more explicit.

Reply:
OK, thanks.